### PR TITLE
added support for incremental predicates in delete+insert strategy

### PR DESCRIPTION
**Pull request by VarunSharma15**
### Description

Added support for incremental_predicates in delete+insert strategy and added testcases for the same.

image

### Checklist
 - [ ] I have run this code in development and it appears to resolve the stated issue
 - [ ] This PR includes tests, or tests are not required/relevant for this PR
 - [ ] I have updated the `CHANGELOG.md` with information about my change

### CODE REVIEW

**Review by rohitagr06 — APPROVED** [review]
LGTM

**Review by SatishChGit — CHANGES_REQUESTED** [review]
needs to address a comment other wise looks good

File: .github/workflows/ci-integration-tests.yml [hunk 1]
```diff
@@ -10,7 +10,7 @@ jobs:
   test:
     strategy:
       matrix:
-        python: ['3.7', '3.8', '3.9','3.10', '3.11']
+        python: ['3.7.16', '3.8', '3.9','3.10', '3.11']
```
[the remark this answers is not in the document]
Comment (VarunSharma15) [reply .github/workflows/ci-integration-tests.yml:13]: Rohit already added the comment, so I didn't.
This is not conflicting.

[the remark this answers is not in the document]
Comment (VarunSharma15) [reply .github/workflows/ci-integration-tests.yml:13]: There is an error popping up with python 3.7.17 regarding bz2 dependency.
It is working fine upto 3.7.16. Once there is a new release for python 3.7 we can revert this back 3.7.16 to 3.7
Also we will mention this in release note.

**Review by SatishChGit — APPROVED** [review]

**Pull request merged**